### PR TITLE
review: test: add tests for CtStatementList

### PR DESCRIPTION
**Pull request by Rohitesh-Kumar-Jain**
Hi, I have added new tests for the class `CtStatementList`, I had written them earlier but to avoid working on conflicting branches I am pushing them now, this PR is similar to #4082, so I will automatically implement changes based on reviews on that PR.

These tests are intended to kill these mutations

1)

Screenshot 2021-08-11 at 9 58 10 AM

2)

Screenshot 2021-08-11 at 9 58 25 AM

3)
Screenshot 2021-08-11 at 9 58 38 AM

4)
Screenshot 2021-08-11 at 9 58 56 AM

Link to #3967

### CODE REVIEW

File: src/test/java/spoon/test/ctStatementList/CtStatementListTest.java [hunk 1]
```diff
@@ -1,35 +1,52 @@
...
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.CoreMatchers.is;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 public class CtStatementListTest {
+    private static Factory factory;
+
+    private static CtStatementList getStatementListInitializedWithOneStatement() {
+        factory = new Launcher().getFactory();
```
Comment (slarse) [inline src/test/java/spoon/test/ctStatementList/CtStatementListTest.java:34]: Setting the `factory` field is a very unexpected side-effect of this method. I would in general avoid mutable fields in test classes (or any classes, for that matter) when possible. If a new test is added that uses this field without first calling the associated method, there could be an NPE if that test is executed first. Tests that depend on the order of test execution is a big nono.

If you really want this as a field, then set it up in an `@BeforeEach` method, but make it an instance field and not static. I'd personally avoid even that, though, and just create the factory in each test that needs it.

Just create the factory in each test instead.

```suggestion
    private static CtStatementList getStatementListInitializedWithOneStatement() {
        Factory factory = new Launcher().getFactory();
```

**Review by slarse — CHANGES_REQUESTED** [review]
See comment

File: src/test/java/spoon/test/ctStatementList/CtStatementListTest.java (hunk 1 above)
Comment (Rohitesh-Kumar-Jain) [reply src/test/java/spoon/test/ctStatementList/CtStatementListTest.java:34]: > Just create the factory in each test instead.

Thanks, I will do the same

**Review by slarse — APPROVED** [review]
LGTM, thanks @Rohitesh-Kumar-Jain!

**Pull request merged**